javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserEnumDeclaration.java

```diff
@@ -226,16 +226,65 @@ public int hashCode() {
         ancestors.add(enumClass);
         if (wrappedNode.getImplementedTypes() != null) {
             for (ClassOrInterfaceType implementedType : wrappedNode.getImplementedTypes()) {
-                SymbolReference<ResolvedTypeDeclaration> implementedDeclRef = new SymbolSolver(typeSolver).solveTypeInType(this, implementedType.getName().getId());
-                if (!implementedDeclRef.isSolved() && !acceptIncompleteList) {
-                    throw new UnsolvedSymbolException(implementedType.getName().getId());
+                ResolvedReferenceType ancestor;
+                try {
+                    ancestor = toReferenceType(implementedType);
```

Comment:
perhaps we can do the add to ancestors directly here, avoiding the temp var and the null assignment